tasks/akeyless/src/main/java/com/walmartlabs/concord/plugins/akeyless/model/TaskParamsImpl.java

```diff
@@ -138,6 +143,49 @@ public Action action() {
         return action(input);
     }
 
+    /**
+     * @param o Input value with may be a String or Map of Concord secret
+     *          info (org, name, password)
+     * @param secretExporter for access Concord's Secrets API
+     * @return String value from direct input or exported Secret value
+     */
+    @SuppressWarnings("unchecked")
+    private static String stringOrSecret(Object o, SecretExporter secretExporter) {
+        if (o instanceof String) {
+            return (String) o;
+        }
+
+        if (! (o instanceof Map)) {
+            throw new IllegalArgumentException("Invalid data type given for sensitive argument. Must be string or map.");
+        }
+
+        ((Map<?, ?>) o).forEach((key, value) -> {
+            if (!(key instanceof String)) {
+                throw new IllegalArgumentException("Non-string key used for secret definition");
+            }
+
+            if (!(value instanceof String)) {
+                throw new IllegalArgumentException("Non-string value used for secret definition");
```

Comment:
maybe add `key` value to the exception message?

Reply:
I doubt very much that this can happen within one process, but still :)

Reply:
That's fair